src/sentry/incidents/models/alert_rule.py

```diff
@@ -585,14 +601,37 @@ class Meta:
 
 
 @register_alert_subscription_callback(AlertRuleMonitorType.ACTIVATED)
-def clean_expired_alerts(subscription: QuerySubscription) -> bool:
+def update_alert_activations(
+    subscription: QuerySubscription, alert_rule: AlertRule, value: float
+) -> bool:
+    activations = alert_rule.activations.filter(
+        finished_at=None, query_subscription_id=subscription.id
+    )
+    for activation in activations:
+        activation.update(metric_value=value)
+
+    return True
+
+
+@register_alert_subscription_callback(AlertRuleMonitorType.ACTIVATED)
+def clean_expired_alerts(
+    subscription: QuerySubscription, alert_rule: AlertRule, value: float
+) -> bool:
     now = timezone.now()
     subscription_end = subscription.date_added + timedelta(
         seconds=subscription.snuba_query.time_window
     )
 
     if now > subscription_end:
         delete_snuba_subscription(subscription)
+        activations = alert_rule.activations.filter(
+            finished_at=None, query_subscription_id=subscription.id
+        )
+        for activation in activations:
+            activation.update(
+                metric_value=value,
+                finished_at=now,
+            )
```

Comment:
If you don't need signals from django you can do an update on the query with `.filter().update()`

Reply:
Oh nice.
but can we iterate through and update on the filter?
Or are you saying the `.update()` will apply to all the instances returned by the filter?

Reply:
>  Or are you saying the .update() will apply to all the instances returned by the filter?

Yes, all matching rows will be updated. Calling `QuerySet.update()` doesn't load any records into memory, instead it emits a query similar to `UPDATE table SET column = ? WHERE ...`